system-x/services/telegram/src/main/java/software/tnb/telegram/service/TelegramBotApi.java

```diff
@@ -0,0 +1,62 @@
+package software.tnb.telegram.service;
+
+import software.tnb.common.account.AccountFactory;
+import software.tnb.common.deployment.WithExternalHostname;
+import software.tnb.common.service.Service;
+import software.tnb.telegram.account.TelegramAccount;
+import software.tnb.telegram.resource.TelegramBotAPIContainer;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.auto.service.AutoService;
+
+import java.util.Map;
+
+@AutoService(TelegramBotApi.class)
+public class TelegramBotApi implements Service, WithExternalHostname {
```

Comment:
`implements WithDockerImage` :cry:

Reply:
also it should implement both local and openshift variants

Reply:
yes, for example, this can't be used in rhoc test suite - as that runs in a pod the testcontainers can't be used (without further configuration). therefore, if possible, you should always split it into local and openshift variants

Reply:
@avano I refactored the service adding both implementation (local, Openshift) https://github.com/tnb-software/TNB/pull/232